analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -121,9 +128,37 @@ public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCod
         // the decision is made on based on whether the project contains the analyzer as NuGet).
         if (context.HasMatchingScope(supportedDiagnostics)
             && context.ShouldAnalyzeTree(sourceTree, generatedCodeRecognizer)
+            && context.Options.ParseSonarLintXmlSettings() is { } sonarLintSettings
+            && IsIncluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
```

Comment:
Also, please do a local analysis with the exclusions or inclusions set, just to see in which format they are put into the sonarlint.xml. If there are spaces after the commas, etc.

Reply:
Result of the local analysis:
`<Property Name="sonar.exclusions">**/*.cs,**/**/*.cs,**/**/**/*.cs</Property>`